CompuCell3D/core/CompuCell3D/Field3D/Dim3D.h

```diff
@@ -44,65 +42,72 @@ namespace CompuCell3D {
     /// Copy constructor
     Dim3D(const Dim3D &dim) : Point3D(dim) {}  
 
-    Dim3D &operator=(const Dim3D pt) {
-      x = pt.x;
-      y = pt.y;
-      z = pt.z;
-      return *this;
-    }
+    Dim3D &operator=(const Dim3D pt);
```

Comment:
I would be curious to see if there is any performance degradation when we move function definitions from header to implementation file possibly sacrificing inlining. I suspect that the effect should be minor or none, but if you have a code handy that can do both styles of implementations maybe we can do quick benchmark?

Reply:
See #447. I measured a pretty significant performance degradation using the regression test suite. But these functions inline in this PR, and the implementations are just below the class definitions to improve readability. So no change now in performance.

Reply:
make sense, I got confused by how github presents changes I thought those functions were in the *.cpp file